vyper/context/types/function.py

```diff
@@ -199,8 +239,15 @@ def from_FunctionDef(
             else:
                 raise StructureException("Bad decorator syntax", decorator)
 
-        if "is_public" not in kwargs:
-            raise FunctionDeclarationException("Visibility must be public or private", node)
+        if "function_visibility" not in kwargs:
+            options = " or ".join(v.value for v in list(FunctionVisibility))
+            raise FunctionDeclarationException(
+                f"Visibility must be set to one of {options}", node,
+            )
+
+        if "state_mutability" not in kwargs:
+            # Assume nonpayable if not set at all (cannot accept Ether, but can modify state)
+            kwargs["state_mutability"] = StateMutability("nonpayable")
```

Comment:
Better to use the enum type
```suggestion
            kwargs["state_mutability"] = StateMutability.NONPAYABLE
```